core/src/test/java/bio/terra/pearl/core/dao/participant/EnrolleeSearchDaoTests.java

```diff
@@ -235,4 +236,100 @@ public void testTaskSearch(TestInfo info) throws Exception {
+    assertThat(result.stream().map(EnrolleeSearchResult::getEnrollee).toList(), containsInAnyOrder(enrollee1, enrollee2));
+
+    // check that a search for a different value returns only the other enrollee
+    facet = new SqlSearchableFacet(
+            new AnswerFacetValue(survey.getStableId(), "question2", 1.5), new AnswerFacetSqlGenerator());
+    result = enrolleeSearchDao.search(studyEnv.getId(), List.of(facet));
+    assertThat(result.stream().map(EnrolleeSearchResult::getEnrollee).toList(), contains(enrollee1));
+
+    // check that a search for a nonexistent value returns no results
+    facet = new SqlSearchableFacet(
+            new AnswerFacetValue(survey.getStableId(), "question1", 0D), new AnswerFacetSqlGenerator());
+    result = enrolleeSearchDao.search(studyEnv.getId(), List.of(facet));
+    assertThat(result.stream().map(EnrolleeSearchResult::getEnrollee).toList(), hasSize(0));
+  }
+
+  @Test
+  @Transactional
+  public void testBooleanAnswerSearch(TestInfo info) {
+    StudyEnvironment studyEnv = studyEnvironmentFactory.buildPersisted(getTestName(info));
+    Enrollee enrollee1 = enrolleeFactory.buildPersisted(getTestName(info), studyEnv);
+    Enrollee enrollee2 = enrolleeFactory.buildPersisted(getTestName(info), studyEnv);
+    Enrollee enrollee3 = enrolleeFactory.buildPersisted(getTestName(info), studyEnv);
+
+    Survey survey = surveyFactory.buildPersisted(getTestName(info));
+    surveyResponseFactory.buildWithAnswers(enrollee1, survey, Map.of("question1", true, "question2", false));
+    surveyResponseFactory.buildWithAnswers(enrollee2, survey, Map.of("question1", true, "question2", true));
+
+    // search should return both enrollees with the same answer
+    SqlSearchableFacet facet = new SqlSearchableFacet(
+            new AnswerFacetValue(survey.getStableId(), "question1", true), new AnswerFacetSqlGenerator());
+    List<EnrolleeSearchResult> result = enrolleeSearchDao.search(studyEnv.getId(), List.of(facet));
+    assertThat(result.stream().map(EnrolleeSearchResult::getEnrollee).toList(), containsInAnyOrder(enrollee1, enrollee2));
+
+    // check that a search for a different value returns only the other enrollee
+    facet = new SqlSearchableFacet(
+            new AnswerFacetValue(survey.getStableId(), "question2", false), new AnswerFacetSqlGenerator());
+    result = enrolleeSearchDao.search(studyEnv.getId(), List.of(facet));
+    assertThat(result.stream().map(EnrolleeSearchResult::getEnrollee).toList(), contains(enrollee1));
+
+    // check that a search for a nonexistent question returns no results
+    facet = new SqlSearchableFacet(
+            new AnswerFacetValue(survey.getStableId(), "question3", false), new AnswerFacetSqlGenerator());
+    result = enrolleeSearchDao.search(studyEnv.getId(), List.of(facet));
+    assertThat(result.stream().map(EnrolleeSearchResult::getEnrollee).toList(), hasSize(0));
+  }
+
+  @Autowired
+  private AnswerService answerService;
 }
```

Comment:
nit: move to top?

Reply:
I'm going to do that in the upcoming PR I'm working on now -- I've got more tests added, so that way it won't merge conflict